test/integration/boundingbox_camera_plugin.cc

```diff
@@ -0,0 +1,474 @@
+  frontBox->SetLocalPosition(frontPosition);
+  frontBox->SetLocalRotation(0, 0, 0);
+  frontBox->SetUserData("label", 2);
+  root->AddChild(frontBox);
+
+  rendering::VisualPtr invisibleBox = scene->CreateVisual();
+  invisibleBox->AddGeometry(scene->CreateBox());
+  invisibleBox->SetOrigin(0.0, 0.0, 0.0);
+  invisibleBox->SetLocalPosition(invisiblePosition);
+  invisibleBox->SetLocalRotation(0, 0, 0);
+  invisibleBox->SetUserData("label", 2);
+  root->AddChild(invisibleBox);
+}
+
+/// \brief Build a scene with 3d oreinted box
+void BuildScene3D(rendering::ScenePtr scene)
+{
+  rendering::VisualPtr root = scene->RootVisual();
+
+  // create front box visual (the smaller box)
+  rendering::VisualPtr box = scene->CreateVisual();
+  box->AddGeometry(scene->CreateBox());
+  box->SetOrigin(0.0, 0.0, 0.0);
+  box->SetLocalPosition(math::Vector3d(2, 0, 0));
+  box->SetLocalRotation(math::Quaternion(1.0, 0.5, 0.5, 0.2));
+  box->SetUserData("label", 1);
+  root->AddChild(box);
+}
+
+/////////////////////////////////////////////////
+void BoundingBoxCameraSensorTest::BoxesWithBuiltinSDF(
+  const std::string &_renderEngine)
+{
+  std::string path = ignition::common::joinPaths(PROJECT_SOURCE_PATH, "test",
+      "integration", "boundingbox_camera_sensor_builtin.sdf");
+  sdf::SDFPtr doc(new sdf::SDF());
+  sdf::init(doc);
+  ASSERT_TRUE(sdf::readFile(path, doc));
+  ASSERT_NE(nullptr, doc->Root());
+  ASSERT_TRUE(doc->Root()->HasElement("model"));
+  auto modelPtr = doc->Root()->GetElement("model");
+  ASSERT_TRUE(modelPtr->HasElement("link"));
+  auto linkPtr = modelPtr->GetElement("link");
+  ASSERT_TRUE(linkPtr->HasElement("sensor"));
+  auto sensorPtr = linkPtr->GetElement("sensor");
+  ASSERT_TRUE(sensorPtr->HasElement("camera"));
+  auto cameraPtr = sensorPtr->GetElement("camera");
+  ASSERT_TRUE(cameraPtr->HasElement("image"));
+  auto imagePtr = cameraPtr->GetElement("image");
+
+  unsigned int width = imagePtr->Get<int>("width");
+  unsigned int height = imagePtr->Get<int>("height");
+
+  EXPECT_EQ(width, 320u);
+  EXPECT_EQ(height, 240u);
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene(scene);
+
+  ignition::sensors::Manager mgr;
+
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(sensorPtr);
+
+  std::string type = sdfSensor.TypeStr();
+  EXPECT_EQ(type, "boundingbox_camera");
+
+  ignition::sensors::BoundingBoxCameraSensor *sensor =
+    mgr.CreateSensor<ignition::sensors::BoundingBoxCameraSensor>(sdfSensor);
+
+  EXPECT_NE(sensor, nullptr);
+  sensor->SetScene(scene);
+
+  EXPECT_EQ(width, sensor->ImageWidth());
+  EXPECT_EQ(height, sensor->ImageHeight());
+
+  auto camera = sensor->BoundingBoxCamera();
+  EXPECT_NE(camera, nullptr);
```

Comment:
```suggestion
  ASSERT_NE(camera, nullptr);
```
Same comment about `ASSERT` instead of `EXPECT` for `nullptr` checks